src/components/application_manager/test/application_manager_impl_test.cc

```diff
@@ -2028,6 +2028,30 @@ TEST_F(ApplicationManagerImplTest, AddAndRemoveQueryAppDevice_SUCCESS) {
   EXPECT_FALSE(app_manager_impl_->IsAppsQueriedFrom(device_handle));
 }
 
+TEST_F(
+    ApplicationManagerImplTest,
+    RegisterApplication_ApplicationMissingInConnections_DoNotAddApplicationToAppList) {
+  smart_objects::SmartObject request_for_registration(
+      smart_objects::SmartType_Map);
+
+  request_for_registration[strings::params][strings::connection_key] =
+      kConnectionKey;
+
+  auto request_for_registration_ptr =
+      std::make_shared<smart_objects::SmartObject>(request_for_registration);
+  std::unique_ptr<plugin_manager::RPCPluginManager> rpc_plugin_manager(
+      new MockRPCPluginManager());
+  app_manager_impl_->SetPluginManager(rpc_plugin_manager);
+
+  ON_CALL(mock_connection_handler_, IsAppPresentInConnections(kConnectionKey))
+      .WillByDefault(Return(false));
+
+  EXPECT_CALL(*mock_policy_handler_, OnAddedNewApplicationToAppList(_, _))
+      .Times(0);
+  auto application =
```

Comment:
@ydementieiev Looks that we need to check the application variable.

Reply:
@AByzhynar you are right, we don't need create it here , I [removed](https://github.com/LuxoftSDL/sdl_core/pull/44/commits/eb9f13417feaa48a59ba6824dfc4becbaf7e3dea#diff-27ad6fe5524d6264e19097c70d82319eR2051)